### DISCUSSION TITLE
我在尝试使用腾讯的edgeone来部署，但是部署失败了

### DISCUSSION THREAD

**Discussion by bakeham** [lobehub/lobehub#5168] [Self Hosting]
使用的部署选项为：
![PixPin_2024-12-24_17-44-20](https://github.com/user-attachments/assets/3bd45ca3-794d-44af-820a-ecee1c36a2ce)
Node.js 版本：22.11.0

报错如下：
`17:21:31.595
Cloning into 'repo_dir'...
17:21:34.616
17:21:34.616
BuildCommit: df0e0aa5442b82dcc05bff977fe3bbfc1f63a216
17:21:34.617
17:21:35.695
Found '/data/workspace/repo_dir/.nvmrc' with version <lts/iron>
17:21:35.911
Downloading and installing node v20.18.1...
17:21:36.077
Downloading https://nodejs.org/dist/v20.18.1/node-v20.18.1-linux-x64.tar.gz...
17:21:36.217
17:21:36.324
##########                                                                13.9%
17:21:36.446
#####################################                                     51.6%
17:21:36.508
###########################################################               83.1%
17:21:36.508
######################################################################## 100.0%
17:21:36.549
Computing checksum with sha256sum
17:21:36.689
Checksums matched!
17:21:38.198
Now using node v20.18.1 (npm v10.8.2)
17:21:38.231
Running npm install
17:21:52.386
npm error code ERESOLVE
17:21:52.387
npm error ERESOLVE unable to resolve dependency tree
17:21:52.387
npm error
17:21:52.387
npm error While resolving: @lobehub/chat@1.39.0
17:21:52.387
npm error Found: react@19.0.0
17:21:52.387
npm error node_modules/react
17:21:52.387
npm error   react@"^19.0.0" from the root project
17:21:52.387
npm error
17:21:52.387
npm error Could not resolve dependency:
17:21:52.387
npm error peer react@"^16.8.0 || ^17 || ^18" from @codesandbox/sandpack-react@2.19.10
17:21:52.387
npm error node_modules/@codesandbox/sandpack-react
17:21:52.387
npm error   @codesandbox/sandpack-react@"^2.19.9" from the root project
17:21:52.387
npm error
17:21:52.387
npm error Fix the upstream dependency conflict, or retry
17:21:52.387
npm error this command with --force or --legacy-peer-deps
17:21:52.387
npm error to accept an incorrect (and potentially broken) dependency resolution.
17:21:52.387
npm error
17:21:52.387
npm error
17:21:52.387
npm error For a full report see:
17:21:52.387
npm error /data/home/.npm/_logs/2024-12-24T09_21_38_307Z-eresolve-report.txt
17:21:52.390
npm notice
17:21:52.390
npm notice New major version of npm available! 10.8.2 -> 11.0.0
17:21:52.390
npm notice Changelog: https://github.com/npm/cli/releases/tag/v11.0.0
17:21:52.390
npm notice To update run: npm install -g npm@11.0.0
17:21:52.390
npm notice
17:21:52.390
npm error A complete log of this run can be found in: /data/home/.npm/_logs/2024-12-24T09_21_38_307Z-debug-0.log
17:21:52.415
build failed.`

**Comment by arvinxx** [comment]
使用pnpm i

**Reply by bakeham** [reply]
十分感谢您的回复！还请您再帮我看下新的情况：
成功了，但似乎卡在这里了，等了很久判部署失败了，
```
 WARN  Issues with peer dependencies found
10:21:58.579
.
10:21:58.579
├─┬ @lobehub/ui 1.154.1
10:21:58.579
│ ├─┬ @react-spring/web 9.7.5
10:21:58.579
│ │ ├── ✕ unmet peer react@"^16.8.0 || ^17.0.0 || ^18.0.0": found 19.0.0
10:21:58.579
│ │ ├── ✕ unmet peer react-dom@"^16.8.0 || ^17.0.0 || ^18.0.0": found 19.0.0
10:21:58.579
│ │ ├─┬ @react-spring/animated 9.7.5
10:21:58.579
│ │ │ ├── ✕ unmet peer react@"^16.8.0 || ^17.0.0 || ^18.0.0": found 19.0.0
10:21:58.579
│ │ │ └─┬ @react-spring/shared 9.7.5
10:21:58.579
│ │ │   └── ✕ unmet peer react@"^16.8.0 || ^17.0.0 || ^18.0.0": found 19.0.0
10:21:58.579
│ │ └─┬ @react-spring/core 9.7.5
10:21:58.579
│ │   └── ✕ unmet peer react@"^16.8.0 || ^17.0.0 || ^18.0.0": found 19.0.0
10:21:58.579
│ ├─┬ ahooks 3.8.4
10:21:58.579
│ │ └── ✕ unmet peer react@"^16.8.0 || ^17.0.0 || ^18.0.0": found 19.0.0
10:21:58.579
│ ├─┬ @emoji-mart/react 1.1.1
10:21:58.579
│ │ └── ✕ unmet peer react@"^16.8 || ^17 || ^18": found 19.0.0
10:21:58.579
│ ├─┬ @giscus/react 3.0.0
10:21:58.579
│ │ ├── ✕ unmet peer react@"^16 || ^17 || ^18": found 19.0.0
10:21:58.579
│ │ └── ✕ unmet peer react-dom@"^16 || ^17 || ^18": found 19.0.0
10:21:58.579
│ └─┬ react-avatar-editor 13.0.2
10:21:58.579
│   ├── ✕ unmet peer react@"^0.14.0 || ^16.0.0 || ^17.0.0 || ^18.0.0": found 19.0.0
10:21:58.579
│   └── ✕ unmet peer react-dom@"^0.14.0 || ^16.0.0 || ^17.0.0 || ^18.0.0": found 19.0.0
10:21:58.579
├─┬ ink 4.4.1
10:21:58.579
│ └─┬ react-reconciler 0.29.2
10:21:58.579
│   └── ✕ unmet peer react@^18.3.1: found 19.0.0
10:21:58.579
├─┬ @lobehub/i18n-cli 1.20.2
10:21:58.579
│ └─┬ zustand 4.5.5
10:21:58.579
│   └─┬ use-sync-external-store 1.2.2
10:21:58.579
│     └── ✕ unmet peer react@"^16.8.0 || ^17.0.0 || ^18.0.0": found 19.0.0
10:21:58.580
├─┬ @codesandbox/sandpack-react 2.19.10
10:21:58.580
│ ├── ✕ unmet peer react@"^16.8.0 || ^17 || ^18": found 19.0.0
10:21:58.580
│ └── ✕ unmet peer react-dom@"^16.8.0 || ^17 || ^18": found 19.0.0
10:21:58.580
├─┬ @icons-pack/react-simple-icons 9.6.0
10:21:58.580
│ └── ✕ unmet peer react@"^16.13 || ^17 || ^18": found 19.0.0
10:21:58.580
├─┬ @sentry/nextjs 7.120.2
10:21:58.580
│ ├── ✕ unmet peer react@"16.x || 17.x || 18.x": found 19.0.0
10:21:58.580
│ ├── ✕ unmet peer next@"^10.0.8 || ^11.0 || ^12.0 || ^13.0 || ^14.0": found 15.1.2
10:21:58.580
│ └─┬ @sentry/react 7.120.2
10:21:58.580
│   └── ✕ unmet peer react@"15.x || 16.x || 17.x || 18.x": found 19.0.0
10:21:58.580
├─┬ next-mdx-remote 4.4.1
10:21:58.580
│ ├── ✕ unmet peer react@">=16.x <=18.x": found 19.0.0
10:21:58.580
│ └── ✕ unmet peer react-dom@">=16.x <=18.x": found 19.0.0
10:21:58.580
├─┬ react-confetti 6.1.0
10:21:58.580
│ └── ✕ unmet peer react@"^16.3.0 || ^17.0.1 || ^18.0.0": found 19.0.0
10:21:58.580
├─┬ react-lazy-load 4.0.1
10:21:58.580
│ ├── ✕ unmet peer react@"^17.0.0 || ^18.0.0": found 19.0.0
10:21:58.580
│ └── ✕ unmet peer react-dom@"^17.0.0 || ^18.0.0": found 19.0.0
10:21:58.580
├─┬ @lobehub/lint 1.24.4
10:21:58.580
│ ├─┬ @umijs/lint 4.4.2
10:21:58.580
│ │ └─┬ stylelint-config-standard 25.0.0
10:21:58.580
│ │   ├── ✕ unmet peer stylelint@^14.4.0: found 15.11.0
10:21:58.580
│ │   └─┬ stylelint-config-recommended 7.0.0
10:21:58.580
│ │     └── ✕ unmet peer stylelint@^14.4.0: found 15.11.0
10:21:58.580
│ ├─┬ stylelint-config-recommended 14.0.1
10:21:58.580
│ │ └── ✕ unmet peer stylelint@^16.1.0: found 15.11.0
10:21:58.580
│ ├─┬ stylelint-less 3.0.1
10:21:58.580
│ │ └── ✕ unmet peer stylelint@^16.0.2: found 15.11.0
10:21:58.580
│ └─┬ semantic-release-config-gitmoji 1.5.3
10:21:58.580
│   └─┬ @achingbrain/semantic-release-github 0.0.2
10:21:58.580
│     └─┬ @octokit/plugin-throttling 3.7.0
10:21:58.580
│       └── ✕ unmet peer @octokit/core@^3.5.0: found 5.2.0
10:21:58.580
├─┬ eslint-plugin-unused-imports 4.0.1
10:21:58.580
│ ├── ✕ unmet peer eslint@^9.0.0: found 8.57.1
10:21:58.581
│ └── ✕ unmet peer @typescript-eslint/eslint-plugin@^8.0.0-0: found 7.5.0
10:21:58.581
├─┬ @lobehub/chat-plugin-sdk 1.32.4
10:21:58.581
│ └─┬ swagger-client 3.32.2
10:21:58.581
│   └─┬ @swagger-api/apidom-reference 1.0.0-beta.5
10:21:58.581
│     └─┬ @swagger-api/apidom-parser-adapter-api-design-systems-json 1.0.0-beta.5
10:21:58.581
│       └─┬ @swagger-api/apidom-parser-adapter-json 1.0.0-beta.5
10:21:58.581
│         └─┬ tree-sitter-json 0.24.8
10:21:58.581
│           └── ✕ unmet peer tree-sitter@^0.21.1: found 0.22.1
10:21:58.581
└─┬ nuqs 1.20.0
10:21:58.586
  └── ✕ unmet peer next@">=13.4 <14.0.2 || ^14.0.3": found 15.1.2
10:21:58.586
10:21:58.586
Done in 1m 37.4s
10:21:58.668
Running npm run build
10:21:58.851
10:21:58.851
> @lobehub/chat@1.39.0 build
10:21:58.851
> next build
10:21:58.851
10:22:00.788
Attention: Next.js now collects completely anonymous telemetry regarding usage.
10:22:00.788
This information is used to shape Next.js' roadmap and prioritize features.
10:22:00.788
You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
10:22:00.788
https://nextjs.org/telemetry
10:22:00.788
10:22:00.922
   ▲ Next.js 15.1.2
10:22:00.922
10:22:00.995
   Creating an optimized production build ...
10:22:01.511
 ○ (serwist) The service worker will not be automatically registered, please call 'window.serwist.register()' in 'componentDidMount' or 'useEffect'.
10:22:01.519
 ✓ (serwist) Bundling the service worker script with the URL '/sw.js' and the scope '/'...
```

**Reply by SpeedupMaster** [reply]
> 十分感谢您的回复！还请您再帮我看下新的情况： 成功了，但似乎卡在这里了，等了很久判部署失败了，
> 
> ```
>  WARN  Issues with peer dependencies found
> 10:21:58.579
> .
> 10:21:58.579
> ├─┬ @lobehub/ui 1.154.1
> 10:21:58.579
> │ ├─┬ @react-spring/web 9.7.5
> 10:21:58.579
> │ │ ├── ✕ unmet peer react@"^16.8.0 || ^17.0.0 || ^18.0.0": found 19.0.0
> 10:21:58.579
> │ │ ├── ✕ unmet peer react-dom@"^16.8.0 || ^17.0.0 || ^18.0.0": found 19.0.0
> 10:21:58.579
> │ │ ├─┬ @react-spring/animated 9.7.5
> 10:21:58.579
> │ │ │ ├── ✕ unmet peer react@"^16.8.0 || ^17.0.0 || ^18.0.0": found 19.0.0
> 10:21:58.579
> │ │ │ └─┬ @react-spring/shared 9.7.5
> 10:21:58.579
> │ │ │   └── ✕ unmet peer react@"^16.8.0 || ^17.0.0 || ^18.0.0": found 19.0.0
> 10:21:58.579
> │ │ └─┬ @react-spring/core 9.7.5
> 10:21:58.579
> │ │   └── ✕ unmet peer react@"^16.8.0 || ^17.0.0 || ^18.0.0": found 19.0.0
> 10:21:58.579
> │ ├─┬ ahooks 3.8.4
> 10:21:58.579
> │ │ └── ✕ unmet peer react@"^16.8.0 || ^17.0.0 || ^18.0.0": found 19.0.0
> 10:21:58.579
> │ ├─┬ @emoji-mart/react 1.1.1
> 10:21:58.579
> │ │ └── ✕ unmet peer react@"^16.8 || ^17 || ^18": found 19.0.0
> 10:21:58.579
> │ ├─┬ @giscus/react 3.0.0
> 10:21:58.579
> │ │ ├── ✕ unmet peer react@"^16 || ^17 || ^18": found 19.0.0
> 10:21:58.579
> │ │ └── ✕ unmet peer react-dom@"^16 || ^17 || ^18": found 19.0.0
> 10:21:58.579
> │ └─┬ react-avatar-editor 13.0.2
> 10:21:58.579
> │   ├── ✕ unmet peer react@"^0.14.0 || ^16.0.0 || ^17.0.0 || ^18.0.0": found 19.0.0
> 10:21:58.579
> │   └── ✕ unmet peer react-dom@"^0.14.0 || ^16.0.0 || ^17.0.0 || ^18.0.0": found 19.0.0
> 10:21:58.579
> ├─┬ ink 4.4.1
> 10:21:58.579
> │ └─┬ react-reconciler 0.29.2
> 10:21:58.579
> │   └── ✕ unmet peer react@^18.3.1: found 19.0.0
> 10:21:58.579
> ├─┬ @lobehub/i18n-cli 1.20.2
> 10:21:58.579
> │ └─┬ zustand 4.5.5
> 10:21:58.579
> │   └─┬ use-sync-external-store 1.2.2
> 10:21:58.579
> │     └── ✕ unmet peer react@"^16.8.0 || ^17.0.0 || ^18.0.0": found 19.0.0
> 10:21:58.580
> ├─┬ @codesandbox/sandpack-react 2.19.10
> 10:21:58.580
> │ ├── ✕ unmet peer react@"^16.8.0 || ^17 || ^18": found 19.0.0
> 10:21:58.580
> │ └── ✕ unmet peer react-dom@"^16.8.0 || ^17 || ^18": found 19.0.0
> 10:21:58.580
> ├─┬ @icons-pack/react-simple-icons 9.6.0
> 10:21:58.580
> │ └── ✕ unmet peer react@"^16.13 || ^17 || ^18": found 19.0.0
> 10:21:58.580
> ├─┬ @sentry/nextjs 7.120.2
> 10:21:58.580
> │ ├── ✕ unmet peer react@"16.x || 17.x || 18.x": found 19.0.0
> 10:21:58.580
> │ ├── ✕ unmet peer next@"^10.0.8 || ^11.0 || ^12.0 || ^13.0 || ^14.0": found 15.1.2
> 10:21:58.580
> │ └─┬ @sentry/react 7.120.2
> 10:21:58.580
> │   └── ✕ unmet peer react@"15.x || 16.x || 17.x || 18.x": found 19.0.0
> 10:21:58.580
> ├─┬ next-mdx-remote 4.4.1
> 10:21:58.580
> │ ├── ✕ unmet peer react@">=16.x <=18.x": found 19.0.0
> 10:21:58.580
> │ └── ✕ unmet peer react-dom@">=16.x <=18.x": found 19.0.0
> 10:21:58.580
> ├─┬ react-confetti 6.1.0
> 10:21:58.580
> │ └── ✕ unmet peer react@"^16.3.0 || ^17.0.1 || ^18.0.0": found 19.0.0
> 10:21:58.580
> ├─┬ react-lazy-load 4.0.1
> 10:21:58.580
> │ ├── ✕ unmet peer react@"^17.0.0 || ^18.0.0": found 19.0.0
> 10:21:58.580
> │ └── ✕ unmet peer react-dom@"^17.0.0 || ^18.0.0": found 19.0.0
> 10:21:58.580
> ├─┬ @lobehub/lint 1.24.4
> 10:21:58.580
> │ ├─┬ @umijs/lint 4.4.2
> 10:21:58.580
> │ │ └─┬ stylelint-config-standard 25.0.0
> 10:21:58.580
> │ │   ├── ✕ unmet peer stylelint@^14.4.0: found 15.11.0
> 10:21:58.580
> │ │   └─┬ stylelint-config-recommended 7.0.0
> 10:21:58.580
> │ │     └── ✕ unmet peer stylelint@^14.4.0: found 15.11.0
> 10:21:58.580
> │ ├─┬ stylelint-config-recommended 14.0.1
> 10:21:58.580
> │ │ └── ✕ unmet peer stylelint@^16.1.0: found 15.11.0
> 10:21:58.580
> │ ├─┬ stylelint-less 3.0.1
> 10:21:58.580
> │ │ └── ✕ unmet peer stylelint@^16.0.2: found 15.11.0
> 10:21:58.580
> │ └─┬ semantic-release-config-gitmoji 1.5.3
> 10:21:58.580
> │   └─┬ @achingbrain/semantic-release-github 0.0.2
> 10:21:58.580
> │     └─┬ @octokit/plugin-throttling 3.7.0
> 10:21:58.580
> │       └── ✕ unmet peer @octokit/core@^3.5.0: found 5.2.0
> 10:21:58.580
> ├─┬ eslint-plugin-unused-imports 4.0.1
> 10:21:58.580
> │ ├── ✕ unmet peer eslint@^9.0.0: found 8.57.1
> 10:21:58.581
> │ └── ✕ unmet peer @typescript-eslint/eslint-plugin@^8.0.0-0: found 7.5.0
> 10:21:58.581
> ├─┬ @lobehub/chat-plugin-sdk 1.32.4
> 10:21:58.581
> │ └─┬ swagger-client 3.32.2
> 10:21:58.581
> │   └─┬ @swagger-api/apidom-reference 1.0.0-beta.5
> 10:21:58.581
> │     └─┬ @swagger-api/apidom-parser-adapter-api-design-systems-json 1.0.0-beta.5
> 10:21:58.581
> │       └─┬ @swagger-api/apidom-parser-adapter-json 1.0.0-beta.5
> 10:21:58.581
> │         └─┬ tree-sitter-json 0.24.8
> 10:21:58.581
> │           └── ✕ unmet peer tree-sitter@^0.21.1: found 0.22.1
> 10:21:58.581
> └─┬ nuqs 1.20.0
> 10:21:58.586
>   └── ✕ unmet peer next@">=13.4 <14.0.2 || ^14.0.3": found 15.1.2
> 10:21:58.586
> 10:21:58.586
> Done in 1m 37.4s
> 10:21:58.668
> Running npm run build
> 10:21:58.851
> 10:21:58.851
> > @lobehub/chat@1.39.0 build
> 10:21:58.851
> > next build
> 10:21:58.851
> 10:22:00.788
> Attention: Next.js now collects completely anonymous telemetry regarding usage.
> 10:22:00.788
> This information is used to shape Next.js' roadmap and prioritize features.
> 10:22:00.788
> You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
> 10:22:00.788
> https://nextjs.org/telemetry
> 10:22:00.788
> 10:22:00.922
>    ▲ Next.js 15.1.2
> 10:22:00.922
> 10:22:00.995
>    Creating an optimized production build ...
> 10:22:01.511
>  ○ (serwist) The service worker will not be automatically registered, please call 'window.serwist.register()' in 'componentDidMount' or 'useEffect'.
> 10:22:01.519
>  ✓ (serwist) Bundling the service worker script with the URL '/sw.js' and the scope '/'...
> ```

这里好像确实会很久，七八分钟吧

**Reply by bakeham** [reply]
执行到
` ✓ (serwist) Bundling the service worker script with the URL '/sw.js' and the scope '/'...
`
5分钟后，被腾讯edgeOne判定为构建失败
T^T

**Reply by bakeham** [reply]
> 
> 这里好像确实会很久，七八分钟吧

目前的配置为：
![PixPin_2024-12-25_11-22-42](https://github.com/user-attachments/assets/7885a5ed-1595-43f1-8e91-45e2a01056f9)

**Reply by arvinxx** [reply]
你要找放大构建时间的地方。5分钟不够完成构建的

**Reply by bakeham** [reply]
感谢大家的回复，目前构建时间为1243秒，咨询了官方技术支持，反馈为：最大构建超时为20分钟，请问有办法提高构建速度吗？比如构建时携带一些附加参数？

**Reply by dislazy** [reply]
请问大佬最后部署成功了吗
